### PR TITLE
Refactor machinery hand attack

### PR DESCRIPTION
**Pull request by m-dzianishchyts**
## What Does This PR Do
Extract preprocessing of machinery hand attack.

## Why It's Good For The Game
This will make it easier for coders to modify machinery interaction limitations without duplicate code.

## Testing
Works fine.

### CODE REVIEW

File: code/game/machinery/machinery.dm [hunk 1]
```diff
@@ -222,6 +222,15 @@
 		return attack_hand(user)
 
 /obj/machinery/attack_hand(mob/user as mob)
+	var/attack_result = try_attack_hand(user)
+	if (attack_result != null)
```
Comment (DGamerL) [inline code/game/machinery/machinery.dm:226]:
```suggestion
	if(attack_result != null)
```

File: code/game/machinery/machinery.dm [hunk 2]
```diff
@@ -245,10 +254,6 @@
 	if(!interact_offline && stat & (NOPOWER|BROKEN|MAINT))
 		return TRUE
```
Comment (Sirryan2002) [inline code/game/machinery/machinery.dm:256]: please document these procs really well, coders should really understand why and when to use `try_attack_hand`

File: code/game/machinery/machinery.dm [hunk 3]
```diff
@@ -222,6 +222,25 @@
...
+  *
+  * If overriding and extending interaction limitations, better call this with ..()
+  * unless you really know what you are doing.
+  *
+  * Returns TRUE when interaction is done due to different limitations and nothing should be done next.
+  * Returns FALSE when interaction can be continued.
+  * Arguments:
+  * * user - the mob interacting with this machinery
+  */
+/obj/machinery/proc/try_attack_hand(mob/user as mob)
```
Comment (Gaxeer) [inline code/game/machinery/machinery.dm:243]: as is redundant in procs

Comment (GDNgit) [inline code/game/machinery/machinery.dm:243]:
```suggestion
/obj/machinery/proc/try_attack_hand(mob/user)
```
The AS keyword is only useful for verbs/lists using as anything

**Review by GDNgit — CHANGES_REQUESTED** [review]

**Review by GDNgit — APPROVED** [review]
Looks fine codewise. No need to have a CI though because this isn't playerfacing.

**Review by DGamerL — APPROVED** [review]
LGTM

**Review by lewcc — APPROVED** [review]

**Pull request merged**